reference/tax_settings.v3.yml

```diff
@@ -170,6 +170,9 @@ components:
             - FIXED
             - BASIC
             - DISABLE
+        should_subtract_store_tax:
+          type: boolean
+          description: When enabled, we avoid rounding during subtract store tax processes.
```

Comment:
```suggestion
          description: If a merchant enters **tax-inclusive** prices, this setting indicates whether a store will update the **product's list price** on the storefront to reflect the shopper's tax zone. When enabled, the store subtracts the tax rate  before calculating tax using the shopper's tax zone. When disabled, the tax-inclusive price is the same across all tax zones, and the store updates the **product base price** to reflect differences in tax rates across zones.
```

Reply:
Need someone to check the accuracy of my statement

Reply:
i'm not sure how `and the store updates the **product base price** to reflect differences in tax rates across zones` works with tax-inclusive pricing. Surely it's not an update to the catalog product object.

Reply:
Thanks @bc-andreadao. Certainly the original description isn't accurate - any rounding concerns are a separate matter from this feature, which is simply whether or not we subtract store tax.

I'm not sure I would speak to specific implementation or describe it the way above. How about some minor adjustments:

> This setting applies only if a merchant enters **tax-inclusive** prices.

👍 I think it's important to point this out - very nice.  

It may also be important to point out that it's only relevant to basic tax - but this may introduce further descriptive complexity since basic tax is always used for **tax pricing** but can optionally be used for **tax quotation**.

> When enabled, the store subtracts the item's store tax rate before calculating tax using the shopper's tax zone. 

I've adjusted the wording here slightly. A few notes:
* **Item** instead of using **Product** given this approach applies to non-product items too (e.g. shipping, wrapping).
* **Store tax rate** to hopefully more descriptively answer "Which tax rate?" (instead of just "the tax rate").

Alternatively:

> When enabled, the store subtracts the item's store tax rate before calculating tax using the shopper's tax zone. The tax-exclusive amount will be the same across all tax zones.

This version describes the tax exclusive amount being consistent, to hopefully better compare the two options.

> When disabled, the tax-inclusive price remains the same across all tax zones, only the tax amount will vary based on the shopper's location. The tax-exclusive amount may vary between tax zones.

So... perhaps...

> This setting applies only if a merchant enters **tax-inclusive** prices. When enabled, the store subtracts the item's store tax rate before calculating tax using the shopper's tax zone. The tax-exclusive amount will be the same across all tax zones. When disabled, the tax-inclusive price remains the same across all tax zones, only the tax amount will vary based on the shopper's location. The tax-exclusive amount may vary between tax zones.

Reply:
Thanks @theromulans, I found that super helpful ! 

> It may also be important to point out that it's only relevant to basic tax - but this may introduce further descriptive complexity since basic tax is always used for **tax pricing** but can optionally be used for **tax quotation**.
 
Would it be accurate (and simpler) if I flipped this and said that the setting "doesn't apply if a merchant uses a third-party tax provider for tax quotations"? I was looking at this [slack conversation](https://bigcommerce.slack.com/archives/C44T5JWP2/p1679311143545369?thread_ts=1674643606.762549&cid=C44T5JWP2), but wasn't sure what the final conclusion was.  

Could you explain what you mean when you say that there would be descriptive complexity if I mentioned basic tax?

Also, can the `should_subtract_store_tax` setting affect what shoppers see on a storefront **before checkout**? 
For example, if the following were true:
- a merchant enters tax-inclusive prices,
- the storefront displays prices inclusive of tax,
- the tax zone of the shopper's customer group is different than that of the merchant
then do the prices displayed on the product pages depend on the `should_subtract_store_tax` setting?

Reply:
@bc-andreadao Awesome, thanks for your insightful questions as always!

**Tax pricing** happens outside the context of a cart, for individual prices shown on the storefront - these prices are considered estimates or marketing prices. **Tax quotation** is when we have an entire cart context and we are figuring out how much tax the shopper will pay.

All stores always use basic tax for tax pricing... but not all stores use basic tax for tax quotation. This setting is relevant in both cases: tax pricing and tax quotation.

Today the setting can affect the storefront experience, tax pricing. Specifically, when a customer logs in we may assign a different tax zone to them, and hence store tax will be subtracted before the customer group tax zone's rates are added.

In the near future we have our _tax zone price display for storefront_ project that will also utilise GeoIP to determine a better match for an anonymous shopper on the storefront, meaning this setting will have higher relevance for tax pricing.

So the complexity I was speaking to is mainly around getting into the weeds of this behaviour. It's also quite dependant on how the merchant has configured their tax zones. I would safely guess that the majority of stores today do not have tax zones configured in a way that really makes a difference here.

Reply:
Thanks @theromulans ! Just added a sentence about basic tax.

Reply:
Loving it, thanks @bc-andreadao!